weave-js/src/components/PagePanelComponents/Home/Browse3/pages/CallsPage/CallsTableButtons.tsx

```diff
@@ -665,35 +671,77 @@ export const PaginationButtons = () => {
   const start = rowCount > 0 ? page * pageSize + 1 : 0;
   const end = Math.min(rowCount, (page + 1) * pageSize);
 
+  const pageSizes = [10, 25, 50, 100];
+  if (!pageSizes.includes(pageSize)) {
+    pageSizes.push(pageSize);
+    pageSizes.sort((a, b) => a - b);
+  }
+  const pageSizeOptions = pageSizes.map(sz => ({
+    value: sz,
+    label: sz.toString(),
+  }));
+  const pageSizeValue = pageSizeOptions.find(o => o.value === pageSize);
+  const onPageSizeChange = (option: PageSizeOption | null) => {
+    if (option) {
+      apiRef.current.setPageSize(option.value);
+    }
+  };
+
   return (
-    <Box display="flex" alignItems="center" justifyContent="center" padding={1}>
-      <Button
-        variant="ghost"
-        size="medium"
-        onClick={handlePrevPage}
-        disabled={page === 0}
-        icon="chevron-back"
-      />
+    <Box
+      display="flex"
+      alignItems="center"
+      justifyContent="space-between"
+      width="100%"
+      padding={1}>
+      <Box display="flex" alignItems="center">
+        <Button
+          variant="ghost"
+          size="medium"
+          onClick={handlePrevPage}
+          disabled={page === 0}
+          icon="chevron-back"
+        />
+        <Box
+          mx={1}
+          sx={{
+            fontSize: '14px',
+            fontWeight: '400',
+            color: MOON_500,
+            // This is so that when we go from 1-100 -> 101-200, the buttons don't jump
+            minWidth: '90px',
+            display: 'flex',
+            justifyContent: 'center',
+          }}>
+          {start}-{end} of {rowCount}
+        </Box>
+        <Button
+          variant="ghost"
+          size="medium"
+          onClick={handleNextPage}
+          disabled={page >= pageCount - 1}
+          icon="chevron-next"
+        />
+      </Box>
       <Box
-        mx={1}
         sx={{
           fontSize: '14px',
           fontWeight: '400',
           color: MOON_500,
-          // This is so that when we go from 1-100 -> 101-200, the buttons dont jump
-          minWidth: '90px',
           display: 'flex',
-          justifyContent: 'center',
+          alignItems: 'center',
+          gap: '8px',
         }}>
-        {start}-{end} of {rowCount}
+        Per page:
+        <Select<PageSizeOption>
```

Comment:
the selector clips weirdly with the horizontal scroll bar
<img width="174" alt="Screenshot 2025-01-17 at 9 20 28 AM" src="https://github.com/user-attachments/assets/6ce8d8d7-35d4-4713-8975-4aaabfa730a1" />

Reply:
better pic: <img width="183" alt="Screenshot 2025-01-17 at 9 21 46 AM" src="https://github.com/user-attachments/assets/7fb82221-5acc-4cb6-8334-63741b80b27f" />

Reply:
Thanks - I turned on my mac "always show scrollbars" so I hopefully catch these. Added a z-index.